evm/p2p/kademlia.py

```diff
@@ -264,6 +268,22 @@ def neighbours(self, node_id, k=k_bucket_size):
         return sort_by_distance(nodes, node_id)[:k]
 
 
+def binary_get_bucket_for_node(buckets, node):
+    """Return the bucket for a given node."""
+    sorted_buckets = sorted(buckets)
```

Comment:
One of the reasons why @pipermerriam wanted to use binary search here is to avoid unnecessarily iterating over all buckets in get_bucket_for_node(), but this kind of defeats the purpose, no?

Reply:
Yeah, didn't think this was a great way to do it. Is it safe to assume that the list of buckets passed in will always be ordered? (looks kinda like that way with how split_bucket's implemented). Or should I implement a `bisect.insort_left()` somewhere (like split_bucket?) to make sure the list stays ordered?

Reply:
With the current implementation of KBucket.split()/Routing.split_bucket(), yeah, the list will always be sorted, so there's no need for the insort_left(), but this function's docstring should make it clear that it assumes the list is sorted. Although this kind of leaves me wondering if all this extra complexity we're adding is really justified, or if this is just premature optimization that will bite us down the road

Reply:
That's possible, it's hard for me to say. @pipermerriam ?